examples/wireshark.py

```diff
@@ -0,0 +1,160 @@
+"""
+Example script captures network traffic using WireShark and prints a short summary for every protocol.
+Requirements:
+  - Wireshark 2.2.5
+  - pywinauto 0.6.1+
+This example opens "Wireshark", navigates to 'network_connection_name',
+captures network traffic for 'capture_time' seconds, saves all the data to a temporary file,
+parses it and shows a short summer for every protocol.
+"""
+from __future__ import print_function
+from pywinauto.application import Application
+import time
+import csv
+import os
+import sys
+
+def generateDataFile(t_interval, interface_name, file_name):
+    # start Wireshark
+    if (os.path.exists(r"C:\Program Files (x86)\Wireshark")):
+        app = Application(backend='uia').start(r"C:\Program Files (x86)\Wireshark\Wireshark.exe")
+    else:
+        if (os.path.exists(r"C:\Program Files\Wireshark")):
+            app = Application(backend='uia').start(r"C:\Program Files\Wireshark\Wireshark.exe")
+        else:
+            print("Can't find wireshark on your computer")
+
+    win = app['The Wireshark Network Analyzer']
+
+    if app.software_update.exists(timeout=10):
+        app.software_update.skip_this_version.click()
+        app.software_update.wait_not('visible')  # just to make sure it's closed
+    win.wait('ready', timeout=15)
+
+    # Try to find interface_name in TreeView interfaces list
+    tree = win["Interface list Interface list"]
+    interface_name = "\\" + interface_name
+    try:
+        tree.get_item(interface_name).double_click_input()
+    except Exception:
+        print('No such interface')
+        exit()
+
+    win = app['Dialog']
+
+    # Wait while WireShark collect information
+    win.wait('ready')
+    time.sleep(t_interval)
+    win.wait('ready')
+
+    # Stop WireShark
+    win = app['Dialog']
+    win['Stop'].click()
+
+    # open menu (File -> Export Packet Dissections -> As CSV)
+    win['File Alt+F'].select()
+    win = app.top_window()
+    win['Export Packet Dissections'].select()
+    win = app.top_window()
+    win['As CSV...'].click_input()
+
+    # Export FileDialog
+    win = app['Dialog']
+    child = win['Export File Dialog']
+
+    check_file_exist = True
+    try:
+        open(file_name)
+    except IOError:
+        check_file_exist = False
+
+    # input path to temporary file
+    win.type_keys(file_name)
+    child.window(best_match="Save").click()
+    # child['Save'].click()
+
+    # if window "confirm Save As" pop up
+    if (check_file_exist is True):
+        child = win['Confirm Save As']
+        child.window(best_match='yes').click()
+        # child['yes'].click()
+
+    # Quit
+    win['File Alt+F'].select()
+    win = app.top_window()
+    win['Quit Ctrl+Q'].click_input()
+
+    # if window "Quit without Saving" pop up
+    try:
+        win = app.top_window()
```

Comment:
Sorry, missed it previously. Please don't use `top_window()`. Use exact title of possible window like `win = app.ConfirmationTitle` and `win.wait('ready', timeout=2)`. Empty exception handlers (both Exception class and `pass` statements) is unrecommended practice (static analyzer warnings).